.github/workflows/java-maven-openjdk-codeql.yml

```diff
@@ -0,0 +1,68 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        description: |
+          The type of machine to run the job on. Must be provided as a stringified list (e.g. `runs-on: '["ubuntu-latest","self-hosted"]'`)
+        required: true
+        type: string
+      mvn-version:
+        description: |
+          The Maven version used for the execution. You can specify minor or patch version (3.9 or 3.9.1). Default : 3.9
+        required: false
+        type: number
+        default: 3.9
+      mvn-arguments:
+        description: |
+          The extra arguments to pass to the Maven invocation. You can use this to specify a custom profile for example.
+        required: false
+        type: string
+      jdk-version:
+        description: |
+          The JDK version to use for the build.
+        required: true
+        type: number
+
+jobs:
+  analyze-java:
+    name: Analyze Java
+
+    runs-on: ${{ fromJson(inputs.runs-on) }}
+
+    container:
+      image: maven:${{ inputs.mvn-version }}-eclipse-temurin-${{ inputs.jdk-version }}
```

Comment:
I went with temurin in the end since it uses the more common (at least at Coveo) ubuntu linux distro. This ones is keyed to the current LTS (22.04 here)